### PR TITLE
chore: PoC - Define make command to call code generation logic

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

Link to any related issue(s): CLOUDP-279066

Following commands are now supported:
- generates all resources defined in the config file -> `make generate-schema` 
- generate a specific resource -> `make generate-schema resource_name=push_based_log_export` 

Additionally added location of downloaded api spec into gitignore, this way file is available for referencing locally but will cause constant changes in our PRs.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

**Review by EspenAlbert — APPROVED** [review]
Can't wait. Great job!

File: tools/codegen/codespec/string_case.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	"strings"
+)
+
+type SnakeCaseString string
+
+func (snake SnakeCaseString) SnakeCase() string {
+	return string(snake)
+}
+
+func (snake SnakeCaseString) PascalCase() string {
```
Comment (lantoli) [inline tools/codegen/codespec/string_case.go:15]: knit: this is used by our mock library, but I guess you already have everything you need: https://pkg.go.dev/github.com/huandu/xstrings#ToSnakeCase

via https://vektra.github.io/mockery/latest/configuration/#functions

File: tools/codegen/main.go [hunk 2]
```diff
@@ -37,3 +43,19 @@ func getOsArg() *string {
...
+
+func writeToFile(fileName, content string) error {
+	// Open the file with write-only permission, create it if it doesn't exist and truncate its content if it exists
+	file, err := os.OpenFile(fileName, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0o644)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	_, err = file.WriteString(content)
```
Comment (lantoli) [inline tools/codegen/main.go]: knit: similar to this: https://github.com/mongodb/terraform-provider-mongodbatlas/blob/CLOUDP-278652_use_schemas/tools/check-changelog-entry-file/main.go#L35

there is os.WriteFile that might be a bit easier to use

File: tools/codegen/schema/testdata/primitive-attributes.golden.go [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package test_name
+package testname
```
Comment (lantoli) [inline tools/codegen/schema/testdata/primitive-attributes.golden.go:1]: it's idiomatic Go to call the package same as last folder, here it is testdata but testname

**Review by lantoli — APPROVED** [review]

File: tools/codegen/codespec/string_case.go (hunk 1 above)
Comment (AgustinBettati) [reply tools/codegen/codespec/string_case.go:15]: nice, adjusted to use an xstring implementation directly

File: tools/codegen/main.go (hunk 2 above)
Comment (AgustinBettati) [reply tools/codegen/main.go]: thanks, indeed does simplify the code, adjusted

File: tools/codegen/schema/testdata/primitive-attributes.golden.go (hunk 3 above)
Comment (AgustinBettati) [reply tools/codegen/schema/testdata/primitive-attributes.golden.go:1]: This actually a golden file to verify the output of generated code. In the test the resource name is defined as `test_name` so this package is now accurate.

**Pull request merged**